packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java

```diff
@@ -334,17 +332,13 @@ private ResponseEntity<String> doServeEndpoint(String endpointName,
                 LOGGER.error(errorMessage, e);
                 return ResponseEntity.internalServerError().body(errorMessage);
             }
-        } catch (EndpointNotFoundException e) {
-            return ResponseEntity.notFound().build();
-        } catch (EndpointAccessDeniedException e) {
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(
-                    endpointInvoker.createResponseErrorObject(e.getMessage()));
-        } catch (EndpointBadRequestException e) {
-            return ResponseEntity.badRequest().body(
-                    endpointInvoker.createResponseErrorObject(e.getMessage()));
-        } catch (EndpointInternalException e) {
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(
-                    endpointInvoker.createResponseErrorObject(e.getMessage()));
+        } catch (EndpointHttpException e) {
+            var resp = ResponseEntity.status(e.getHttpStatusCode());
```

Comment:
Should we also disallow throwing `2xx` status codes to prevent any abuse of this mechanic? It is not supposed to involve success handling on the client...